### DISCUSSION TITLE
PseudoBulk with condition-aware replicates

### DISCUSSION THREAD

**Discussion by AnjaliC4** [GreenleafLab/ArchR#1260] [Feature Requests]
Hi @rcorces, 

In the creation of pseudobulk replicates, it can happen in a sample-aware manner, which is great. However, it seems like in certain cases where minCells per sample may be insufficient, it creates a pseudobulk across different samples. This can be problematic if the cells from condition A (disease) and mixed with cells from condition B (healthy). This can be a common scenario especially when groupBy="SubClusters", where sub-clusters may not have sufficent cells contributing from every sample. Is it possible to do this analysis in a sample and condition aware manner? Or if you think it would not matter - please do let me know. 

In addition, for my data, arrow files were created per-sample basis which referred to a library and each sample has 2 multiplexed subjects which were later demultiplexed and assigned to their identity in metadata. Is it possible to specify the etadata column corresponding to the demplultiplexed subjects for subject-aware pseudobulking? Can I possibly change the 'Sample' column with my subjects ID for this? Or would I need to change the rows of the metadata/ArchR.arrow files? (since arrow files were created per-sample basis and not subject-basis). 

Thanks for your help. Would really appreciate your input on this. 

Anjali

**Comment by rcorces** [comment]
add a new column to your cellColData that combines sample and condition

**Reply by AnjaliC4** [reply]
Thanks, should this new cellColData column should be named "Sample" ?  Asking because there doesnt seem a way to specify this in addGroupCoverages.

**Reply by rcorces** [reply]
It should be named something different for ex. "sample_condition" and used in the "groupBy" param

**Reply by AnjaliC4** [reply]
Sorry for not being clear. I actually needed groupCoverages by cluster to compare cluster-specific peaks. All I was wondering was whether there was a way to avoid sampling of cells from two different conditions (cells from disease and healthy samples) within a cluster while making pseudobulk replicates.

**Comment by AnjaliC4** [comment]
Hi @rcorces, 

I was hoping to explain my question better. In the pseudobulking procedure you highlight as a flow chart in chapter 9.1, 
1. Do atleast minrep samples have mincells total? Does "sample" here refer to the Sample column of the ArchR project? 
2. If I change this Sample column to my sample-condition (for example), will this change apply when asking if minrep samples (now sample-condition) have mincells total?
3. In generation of groupcoverages, there is an otion: 
useLabels | A boolean value indicating whether to use sample labels to create sample-aware subgroupings during as pseudo-bulk replicate generation.
I belive this refers to "Sample" column in metadata and changing sample column in ArchR project (to sample-condition) should directly change sample-aware to sample-condition aware subgroupings?

Please let me know if my understanding is correct.

Thanks very much!

**Reply by rcorces** [reply]
Sample does refer to the column in Sample. I would ***not*** ever change that column since it is used by many other functions in ArchR. Perhaps you can look at this commit that was made in `release_1.0.2`. This allows you to define "Sample" however you want without changing the `Sample` column
https://github.com/GreenleafLab/ArchR/pull/925/commits/699a886f748066d89fea73db6fd2acd119237f31

**Reply by AnjaliC4** [reply]
This is absolutley awesome and this is exactly what I was hoping for, since I have a multiplexed design. Thank you for implementing this so soon. I have 1.0.2 installed - Do I re-install to implement this change?

Thank  you @rcorces :)

**Reply by rcorces** [reply]
depends on when you last updated 1.0.2. You can check the function on you local machine or just reinstall, restart R, and reload ArchR

**Comment by AnjaliC4** [comment]
What are the recommendations of minCells and minRep to capture biological variability in case of a cluster with more than 50k cells and 50 samples? In your examples, you illustrate that 9.1.1 - sample C is left-out since there were enough to make 5 replicates - is this truly desirable to leave one sample out? And in cases where there are more than 50 samples per cluster - should one make atleast 40 replicates?

Also, how are the pseudobulk replicates used for downstream analysis? This is not clear from documentation. Is it going to be used only for peak calling per replicate per cluster and to add reproducible peak set based on the presence of a peak per replicate? Or will these pseudobulk replicates will be used for any other analysis as well? If peaks are called per pseudobulk replicate - it is desirable to have as many cells as possible per replicate (maxCells) and include all the samples (maxRep) if possible, correct?

Thanks for your time and help.

**Reply by rcorces** [reply]
The answers to these questions are too application specific. There is no single set of parameters that is optimal for every analysis. We've done our best to make pseudobulk generation as biologically correct as possible but users need to assess their data, cell numbers, etc, and make parameter decisions to tailor pseudobulk generation.

Pseudobulk replicates are predominantly used for peak calling but also in footprinting.

**Comment by AnjaliC4** [comment]
Hi @rcorces,

I am having some difficulty understanding all the underlying concepts of pseudo-bulk replication. Let's say, I make two pseudobulk replicates per cluster (one for all cases and other for the control sample). 

In this case, I had to specify sampleLabels ="condition" during addGroupCoverages.  Then when iterative peak merging procedure is performed - the documentation says -  "It is important to note that ArchR uses a normalized metric of significance for peaks to compare the significance of peaks called across different samples. This is because the reported MACS2 significance is proportional to the sequencing depth so peak significance is not immediately comparable across samples."

So in this case, would ArchR still consider per-sample sequencing depth (from Sample column), or would it consider per-condition sequencing depth per cluster? (since I have changed the SampleLabels parameter). 
In addition, reproducibility="2" would be the considering replicability across the two case/control-aware pseudobulk replicates or would it still take into account "Sample" column?

**Reply by rcorces** [reply]
> So in this case, would ArchR still consider per-sample sequencing depth (from Sample column), or would it consider per-condition sequencing depth per cluster? (since I have changed the SampleLabels parameter).

This is a slight typo I guess. Normalization is performed across the pseudo-bulk replicate. Peak calling only uses the pseudobulk replicates, it does not care about samples. This should also answer your second question.
